File: src/main/java/com/blackcowmoo/moomark/auth/model/oauth2/GoogleTokenResult.java
package com.blackcowmoo.moomark.auth.model.oauth2;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString
@NoArgsConstructor
@Getter
public class GoogleTokenResult {
  private String iss;
  private String azp;
  private String aud;
  private String sub;
  private String email;
  @JsonProperty("email_verified")
  private Boolean emailVerified;
  @JsonProperty("at_hash")
  private String atHash;
  private String name;
  private String picture;
  @JsonProperty("given_name")
  private String givenName;
  @JsonProperty("family_name")
  private String familyName;
  private String locale;
  private Integer iat;
  private Integer exp;
}

File: src/main/kotlin/com/blackcowmoo/moomark/auth/model/oauth2/GoogleTokenResult.kt
package com.blackcowmoo.moomark.auth.model.oauth2

import com.fasterxml.jackson.annotation.JsonProperty

data class GoogleTokenResult(
  val iss: String? = null,
  val azp: String? = null,
  val aud: String? = null,
  val sub: String? = null,
  val email: String? = null,
  @JsonProperty("email_verified") val emailVerified: Boolean? = null,
  @JsonProperty("at_hash") val atHash: String? = null,
  val name: String? = null,
  val picture: String? = null,
  @JsonProperty("given_name") val givenName: String? = null,
  @JsonProperty("family_name") val familyName: String? = null,
  val locale: String? = null,
  val iat: Int? = null,
  val exp: Int? = null
)
